### DISCUSSION TITLE
How to query collection entries that have a Replicator block type?

### DISCUSSION THREAD

**Discussion by aaronbushnell** [statamic/cms#8189] [Q&A] [answered]
I'm a bit stuck trying to query all entries that have a particular Replicator block type. Here's what I've tried:
```php
Statamic\Facades\Entry::query()
        ->where('blocks->type', 'hero')
        ->get();
```
Which should yield at least one entry (a handful have a "Hero" block), but I'm getting no results. Am I possibly querying this incorrectly?

**Comment by ryanmitchell** [comment, marked as answer]
Currently you have to query by index... eg `->where('blocks->0->type', 'hero')`.

**Reply by aaronbushnell** [reply]
Thanks! Looks like I’ll have to find another solution because I can’t be certain about the index of the block I’m searching for. Appreciate the help though!

**Reply by duncanmcclean** [reply]
For future reference: once #7454 is merged, you'll be able to do the query without knowing the index. However, there's no ETA on that PR being merged.

**Reply by aaronbushnell** [reply]
Ah, thanks for calling that out, @duncanmcclean! I’ll keep watch of that one when it’s pulled in
